### PR TITLE
[SYCL] Disable vectorization and loop transformation passes

### PR DESCRIPTION
**Pull request by bader**

### CODE REVIEW

File: llvm/lib/Transforms/IPO/PassManagerBuilder.cpp [hunk 1]
```diff
@@ -1034,13 +1045,16 @@ void PassManagerBuilder::addLTOOptimizationPasses(legacy::PassManagerBase &PM) {
...
+  if (!SYCLOptimizationMode) {
+    // Unroll small loops
+    PM.add(createSimpleLoopUnrollPass(OptLevel, DisableUnrollLoops,
+                                      ForgetAllSCEVInLoopUnroll));
+    PM.add(createLoopVectorizePass(true, !LoopVectorize));
+    // The vectorizer may have significantly shortened a loop body; unroll
+    // again.
+    PM.add(createLoopUnrollPass(OptLevel, DisableUnrollLoops,
+                                ForgetAllSCEVInLoopUnroll));
+  }
```
Comment (bader) [inline llvm/lib/Transforms/IPO/PassManagerBuilder.cpp:1057]: TODO: revert this change. This is LTO pipeline, which is not used in SYCL mode.

**Review by MrSidims — APPROVED** [review]

**Comment by bader** [comment]
This PR exposes a regression in `test_stream` from Khronos SYCL-CTS on GPU.
The issue is addressed in https://github.com/intel/compute-runtime/releases/tag/20.39.17972, so we need to update the driver first.

**Comment by bader** [comment]
@mdtoguchi, @AGindinson, I reverted https://github.com/intel/llvm/pull/2402 hoping that changes to optimization pipeline have the effect and this change is not necessary.

Commit with revert: https://github.com/intel/llvm/pull/2458/commits/9e9d053bd7e34763eb15c1f480fa8f9293b6643e

**Comment by bader** [comment]
@DenisBakhvalov, once we removed special flag for ESIMD mode, this patch disables loop optimization transformations for ESIMD mode as well and there are a few failures in ESIMD specific tests. Could suggest the way to fix them?

**Comment by bader** [comment]
> @DenisBakhvalov, once we removed special flag for ESIMD mode, this patch disables loop optimization transformations for ESIMD mode as well and there are a few failures in ESIMD specific tests. Could suggest the way to fix them?

Recent fixes + new GPU driver for Windows resolved all ESIMD specific issues except 3 failing llvm-test-suite tests on Windows (no failures on Linux)

Failed Tests (3):
  SYCL :: ESIMD/private_memory/pm_access_1.cpp
  SYCL :: ESIMD/private_memory/pm_access_2.cpp
  SYCL :: ESIMD/private_memory/pm_access_3.cpp

```
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTSN2cl4sycl6detail19__pf_kernel_wrapperI8KernelIDILi1EEEE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTSN2cl4sycl6detail19__pf_kernel_wrapperI8KernelIDILi1EEEE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTS8KernelIDILi1EE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTS8KernelIDILi1EE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTSN2cl4sycl6detail19__pf_kernel_wrapperI8KernelIDILi2EEEE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTSN2cl4sycl6detail19__pf_kernel_wrapperI8KernelIDILi2EEEE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTSN2cl4sycl6detail19__pf_kernel_wrapperI8KernelIDILi2EEEE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTSN2cl4sycl6detail19__pf_kernel_wrapperI8KernelIDILi2EEEE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTS8KernelIDILi2EE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTS8KernelIDILi2EE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTS8KernelIDILi2EE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTS8KernelIDILi2EE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTSN2cl4sycl6detail19__pf_kernel_wrapperI8KernelIDILi3EEEE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTSN2cl4sycl6detail19__pf_kernel_wrapperI8KernelIDILi3EEEE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTSN2cl4sycl6detail19__pf_kernel_wrapperI8KernelIDILi3EEEE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTSN2cl4sycl6detail19__pf_kernel_wrapperI8KernelIDILi3EEEE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTSN2cl4sycl6detail19__pf_kernel_wrapperI8KernelIDILi3EEEE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTS8KernelIDILi3EE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTS8KernelIDILi3EE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTS8KernelIDILi3EE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTS8KernelIDILi3EE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] warning: GenXPromoteArray: _ZTS8KernelIDILi3EE allocation size is too big: using TPM
[2021-04-28T20:30:26.851Z] LLVM ERROR: Cannot find pointer replacement 
```

@NikitaRudenkoIntel, do know if this a known issue of GPU compiler? I can't find where this diagnostics can be reported in DPC++.

**Comment by NikitaRudenkoIntel** [comment]
Hi, yes there are some issues in TPM. Actually, there is even a ticket with these exact tests and this exact failure. Though it is marked as `implemented`

**Comment by bader** [comment]
> Can you check if your driver is up to date?

We are using 27.20.100.9466 from https://downloadmirror.intel.com/30381/a08/igfx_win10_100.9466.zip.
Is there more recent publicly available version?

**Comment by bader** [comment]
> Hi, yes there are some issues in TPM. Actually, there is even a ticket with these exact tests and this exact failure. It is fixed. Can you check if your driver is up to date?

I managed to get passed this issue with recent optimization pipeline adjustments (probably).
We can get back to the driver version question if we encounter it again.

**Comment by bader** [comment]
I looked at regressions and all of them are issues of external dependencies: OpenCL CPU, Level Zero GPU and llvm-test-suite tests.
I'll address llvm-test-suite test issue as soon as this patch is merged. Low-level runtime issues should be addressed by updating corresponding runtimes.

@mdtoguchi, @AGindinson, @intel/llvm-reviewers-runtime, please, review this change.

**Comment by bader** [comment]
Sorry, didn't notice that there no runtime changes anymore, so I need review only from Mike or Artem.

**Review by mdtoguchi — APPROVED** [review]
Driver OK

**Review by AGindinson — APPROVED** [review]
